lib/fetchers/url.rb

```diff
@@ -91,15 +91,25 @@ def self.transform(target)
 
     attr_reader :files, :archive_path
 
-    def initialize(url, opts)
-      @target = url
+    def initialize(target, opts)
+      if target.is_a?(Hash) && target.key?(:url)
+        @target = target[:url]
+        @upstream_sha256 = target[:sha256]
+      elsif target.is_a?(String)
+        @target = target
+        @upstream_sha256 = nil
+      end
       @insecure = opts['insecure']
       @token = opts['token']
       @config = opts
       @archive_path = nil
       @temp_archive_path = nil
     end
 
+    def upstream_sha256
```

Comment:
You could make this a `attr_reader`:

https://github.com/itmustbejj/inspec/blob/e7ff9dd83ef01e5154ec074183d759bed2ea4b83/lib/fetchers/url.rb#L92